### PR TITLE
Added missing length check that caused an ArgumentNullException (#750)

### LINKED ISSUE #750: ImageProcessingException: An error occurred when processing the image using CropProcessor
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [ ] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
Exception throws when I try to apply EntropyCrop method to this image:
https://yadi.sk/d/nU_mjlfkV5UI2w

Call Stack:
```
SixLabors.ImageSharp.ImageProcessingException: An error occurred when processing the image using EntropyCropProcessor`1. See the inner exception for more detail. ---> SixLabors.ImageSharp.ImageProcessingException: An error occurred when processing the image using CropProcessor`1. See the inner exception for more detail. ---> System.ArgumentNullException: Array cannot be null.
Parameter name: bytes
    at System.Text.Encoding.GetString(Byte* bytes, Int32 byteCount)
   at SixLabors.ImageSharp.EncoderExtensions.GetString(Encoding encoding, ReadOnlySpan`1 buffer)
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifReader.ConvertToString(ReadOnlySpan`1 buffer)
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifReader.ConvertValue(ExifDataType dataType, ReadOnlySpan`1 buffer, UInt32 numberOfComponents)
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifReader.TryReadValue(ExifValue& exifValue)
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifReader.AddValues(List`1 values, UInt32 index)
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifReader.ReadValues()
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifProfile.InitializeValues()
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifProfile.GetValue(ExifTag tag)
   at SixLabors.ImageSharp.Processing.Processors.Transforms.TransformHelpers.UpdateDimensionalMetData[TPixel](Image`1 image)
   at SixLabors.ImageSharp.Processing.Processors.Transforms.TransformProcessorBase`1.AfterImageApply(Image`1 source, Image`1 destination, Rectangle sourceRectangle)
   at SixLabors.ImageSharp.Processing.Processors.CloningImageProcessor`1.CloneAndApply(Image`1 source, Rectangle sourceRectangle)
--- End of inner exception stack trace ---
    at SixLabors.ImageSharp.Processing.Processors.CloningImageProcessor`1.CloneAndApply(Image`1 source, Rectangle sourceRectangle)
   at SixLabors.ImageSharp.Processing.Processors.CloningImageProcessor`1.Apply(Image`1 source, Rectangle sourceRectangle)
   at SixLabors.ImageSharp.Processing.Processors.Transforms.EntropyCropProcessor`1.BeforeImageApply(Image`1 source, Rectangle sourceRectangle)
   at SixLabors.ImageSharp.Processing.Processors.ImageProcessor`1.Apply(Image`1 source, Rectangle sourceRectangle)
--- End of inner exception stack trace ---
    at SixLabors.ImageSharp.Processing.Processors.ImageProcessor`1.Apply(Image`1 source, Rectangle sourceRectangle)
   at SixLabors.ImageSharp.Processing.DefaultInternalImageProcessorContext`1.ApplyProcessor(IImageProcessor`1 processor, Rectangle rectangle)
   at SixLabors.ImageSharp.Processing.DefaultInternalImageProcessorContext`1.ApplyProcessor(IImageProcessor`1 processor)
   at SixLabors.ImageSharp.Processing.EntropyCropExtensions.EntropyCrop[TPixel](IImageProcessingContext`1 source, Single threshold)
```

### Steps to Reproduce
```csharp
            using (var image = Image.Load("img20.jpg"))
            {
                image.Mutate(x => x.EntropyCrop(0.07f));
            }
```

- ImageSharp version: 1.0.0-dev002072
- Other ImageSharp packages and versions:
- Environment (Operating system, version and so on): Win 10
- .NET Framework version: 4.6.2
- Additional information:
img20

### PR DESCRIPTION
**Pull request by dlemstra**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

This patch adds a missing null check and also an extra unit test that checks if the extension method works. This fixes #750.

### CODE REVIEW

**Comment by dlemstra** [comment]
Looks like 2.1 returns an empty string instead.

**Comment by antonfirsov** [comment]
I have a feeling that when we fix such bugs, we always have to add end-to-end regression tests with a test image, even if a unit test has been added based on the root cause.

@JimBobSquarePants @dlemstra am I overthinking?

**Comment by JimBobSquarePants** [comment]
A would add the image from #750 in case we change the underlying implementation and introduce a regressive bug.

File: tests/ImageSharp.Tests/TestImages.cs [hunk 1]
```diff
@@ -160,6 +160,8 @@ public static class Issues
                 public const string OrderedInterleavedProgressive723A = "Jpg/issues/Issue723-Ordered-Interleaved-Progressive-A.jpg";
                 public const string OrderedInterleavedProgressive723B = "Jpg/issues/Issue723-Ordered-Interleaved-Progressive-B.jpg";
                 public const string OrderedInterleavedProgressive723C = "Jpg/issues/Issue723-Ordered-Interleaved-Progressive-C.jpg";
+                public const string ExifGetString750Transform = "Jpg/issues/issue750-exif-tranform.jpg";
```
Comment (dlemstra) [inline tests/ImageSharp.Tests/TestImages.cs]: Small typo here. Thanks for adding these images.

Comment (JimBobSquarePants) [reply tests/ImageSharp.Tests/TestImages.cs]: No worries, I'll update the reference file name later on, will do for now.

**Pull request merged**